yb-voyager/cmd/metadataDB.go

```diff
@@ -342,3 +344,108 @@ func (m *MetaDB) GetExportedEventsStatsForTable(schemaName string, tableName str
 	}
 	return totalCount, inserts, updates, deletes, nil
 }
+
+func (m *MetaDB) GetSegmentsToBeArchived(importCount int) []Segments {
+	var segmentsToArchive []Segments
+	query := fmt.Sprintf(`SELECT segment_no, file_path FROM %s WHERE imported_by_target_db_importer + imported_by_ff_db_importer = ? AND archived = 0 ORDER BY segment_no;`, QUEUE_SEGMENT_META_TABLE_NAME)
+	rows, err := m.db.Query(query, importCount)
+	if err != nil {
+		utils.ErrExit("error while running query on meta db -%s :%v", query, err)
+	}
+	defer func() {
+		err := rows.Close()
+		if err != nil {
+			log.Errorf("failed to close rows: %v", err)
+		}
+	}()
+	for rows.Next() {
+		var segmentNo int64
+		var filePath string
+		err := rows.Scan(&segmentNo, &filePath)
+		if err != nil {
+			utils.ErrExit("error while scanning rows: %v", err)
```

Comment:
Always include error context in the error message e.g. what query did the code execute, what was the objective of the code, etc.